tests/e2e/group/tx.go

```diff
@@ -2173,14 +2173,27 @@ func (s *E2ETestSuite) TestTxLeaveGroup() {
 		),
 	)
 	s.Require().NoError(err, out.String())
-	s.Require().NoError(s.network.WaitForNextBlock())
 
-	out, err = clitestutil.ExecTestCLICmd(clientCtx, client.QueryGroupPoliciesByGroupCmd(), []string{groupID, fmt.Sprintf("--%s=json", flags.FlagOutput)})
+	err = s.network.RetryForBlocks(func() error {
+		out, err = clitestutil.ExecTestCLICmd(clientCtx, client.QueryGroupPoliciesByGroupCmd(), []string{groupID, fmt.Sprintf("--%s=json", flags.FlagOutput)})
+		if err != nil {
+			return err
+		}
+
+		var resp group.QueryGroupPoliciesByGroupResponse
```

Comment:
Which part? The `UnmarshalJSON` and length check? We need that inside to throw an error if it's not what we expect. `ExecTestCLICmd` won't return an error in case that the group policies length is 0.

Reply:
Oh, I see! Makes sense.